cloudevents/emitter.go

```diff
@@ -0,0 +1,60 @@
+package cloudevents
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudevents/sdk-go/v2/client"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/cloudevents/sdk-go/v2/protocol/http"
+	"github.com/cloudevents/sdk-go/v2/types"
+	"github.com/google/uuid"
+)
+
+type Emitter struct {
+	Endpoint    string
+	Source      string
+	Type        string
+	Id          string
+	Data        string
+	ContentType string
+}
+
+func NewEmitter() *Emitter {
+	return &Emitter{
+		Source:      "/boson/fn",
+		Type:        "boson.fn",
+		Id:          uuid.NewString(),
+		Data:        "",
+		ContentType: event.TextPlain,
+	}
+}
+
+func (emitter *Emitter) Emit(endpoint string) (err error) {
+	c, err := newClient(endpoint)
+	if err != nil {
+		return
+	}
+	e := event.Event{
+		Context: event.EventContextV1{
+			Type:   emitter.Type,
+			Source: *types.ParseURIRef(emitter.Source),
+			ID:     emitter.Id,
+		}.AsV1(),
+	}
+	e.SetData(emitter.ContentType, emitter.Data)
+	result := c.Send(context.TODO(), e)
+	if result.Error() != "" {
+		return fmt.Errorf(result.Error())
+	}
+	return nil
+}
+
+func newClient(target string) (c client.Client, err error) {
+	p, err := http.New(http.WithTarget(target))
+	if err != nil {
+		return nil, err
+	}
+	c, err = client.New(p)
+	return
```

Comment:
I might be wrong (it's late) but I think this could be collapsed a little:

```suggestion
	if err != nil {
		return
	}
	return client.New(p)
```